subprojects/base-services/src/main/java/org/gradle/internal/io/LinePerThreadBufferingOutputStream.java

```diff
@@ -67,9 +67,10 @@ public boolean checkError() {
 
     @Override
     public void close() {
-        PrintStream stream = maybeGetStream();
-        if (stream != null) {
-            stream.close();
+        PrintStream currentStream = maybeGetStream();
+        if (currentStream != null) {
+            currentStream.close();
+            stream.set(null);
```

Comment:
Shouldn't this be before the close call?

Reply:
Good point.  Done.